components/channel_layout/channel_identifier_router/actions.ts

```diff
@@ -73,15 +75,26 @@ export function onChannelByIdentifierEnter({match, history}: MatchAndHistory): A
     };
 }
 
-export function getPathFromIdentifier(state: GlobalState, path: string, identifier: string) {
+export async function getPathFromIdentifier(state: GlobalState, path: string, identifier: string) {
     if (path === 'channels') {
         // It's hard to tell an ID apart from a channel name of the same length, so check first if
         // the identifier matches a channel that we have
         const channelsByName = getChannelByName(state, identifier);
         const moreChannelsByName = getOtherChannels(state).find((chan) => chan.name === identifier);
 
         if (identifier.length === LENGTH_OF_ID) {
-            return channelsByName || moreChannelsByName ? 'channel_name' : 'channel_id';
+            if (!channelsByName && !moreChannelsByName) {
+                try {
+                    await Client4.getChannel(identifier);
```

Comment:
This is technically an extra request in this case. I think it's probably fine to do this though since it's an uncommon case

Reply:
@hmhealey I can change it to dispatch an action, it just seemed to be a rare case so I went with just requesting for the data which seemed simpler.

PS: making it an action would need to also change the return type, since actions have `{data: any}` in the signature.